core/src/main/java/org/apache/calcite/sql/SqlLiteral.java

```diff
@@ -284,6 +284,11 @@ public static Comparable value(SqlNode node) {
         return (NlsString) literal.value;
       case NUMERIC:
         return (BigDecimal) literal.value;
+      case INTERVAL_YEAR_MONTH:
+      case INTERVAL_DAY_TIME:
+        final SqlIntervalLiteral.IntervalValue val =
+            (SqlIntervalLiteral.IntervalValue) literal.value;
+        return val.getSign();
```

Comment:
The code path we are observing is: (think of it as stack trace)
SqlLiteral.value() (line: 291)
SqlCallBinding.getOperandLiteralValue() (line: 142)
SqlMonotonicBinaryOperator.getMonotonicity() (line: 77)

So for an expression such as  

cast(...) * (INTERVAL '-1' DAY) 

To make the monotonicity correct, we need to let SqlLiteral.value() (line: 291) return -1. 

Make sense?